CodeGeneration/Sources/generate-swift-syntax/templates/swiftsyntax/RawSyntaxNodesFile.swift

```diff
@@ -270,13 +290,23 @@ func rawSyntaxNodesFile(nodesStartingWith: [Character]) -> SourceFileSyntax {
 
 fileprivate extension Child {
   var rawParameterType: TypeSyntax {
-    let paramType: TypeSyntax
-    if case ChildKind.nodeChoices = kind {
-      paramType = syntaxChoicesType
+    var paramType: TypeSyntax
+    if !kind.isNodeChoicesEmpty {
+      paramType = "\(syntaxChoicesType)"
+    } else if hasBaseType && !isOptional {
+      paramType = "some \(syntaxNodeKind.rawProtocolType)"
```

Comment:
Should we also implement this for optional nodes with a base type using `(some XXXX)?`?

Reply:
If we replace `Raw<Kind>Syntax?` with `(some Raw<Kind>SyntaxNodeProtocol)?`, call sites will no longer be able to just pass `nil` to this parameter because we'll need to specialize the generic type.

A workaround for this is what `XXX<Kind>Syntax` are doing,  e.g.
```swift
type: (some TypeSyntaxProtocol)? = TypeSyntax?.none,
```
I didn't know whether it'd be desirable to change all parameters like this into optional parameters for `RawXXX<Kind>Syntax`, so I've kept them as-is.

Reply:
I see. I think requiring the explicit `nil` for raw syntax makes sense because it prompts you to think whether you should parse that part of the node. Could you add this explanation as a comment?